### PR TITLE
Add TensorScatterUpdate conversion

### PR DESCRIPTION
**Pull request by affranchi**
This change enables "TensorScatterUpdate" operator conversion into ONNX "ScatterND" operator.

### CODE REVIEW

File: keras2onnx/_builtin.py [hunk 1]
```diff
@@ -2187,6 +2187,33 @@ def convert_tf_strided_slice(scope, operator, container):
...
+
+        updates_name = operator.inputs[2].full_name
+        updates = _cal_tensor_value(node.inputs[2])
+        if (updates != None):
+            updates_arr = np.array(updates)
+            updates_tensor = numpy_helper.from_array(updates_arr, operator.inputs[2].full_name + '_value')
+            container.add_initializer_from_tensor(updates_tensor)
+            updates_name = operator.inputs[2].full_name + '_value'
+
+        oopb.add_node_with_output('ScatterND',
```
Comment (wenbingl) [inline keras2onnx/_builtin.py:2211]: thanks for the contribution!
please add op_version=11 if any oopb.add_xxxx function was invoked.

File: keras2onnx/_builtin.py [hunk 2]
```diff
@@ -2187,6 +2187,33 @@ def convert_tf_strided_slice(scope, operator, container):
...
+        node = operator.raw_operator
+
+        indices = _cal_tensor_value(node.inputs[1])
+        indices_arr = np.array(indices)
+        indices_tensor = numpy_helper.from_array(indices_arr, operator.inputs[1].full_name + '_value')
+        container.add_initializer_from_tensor(indices_tensor)
+
+        updates_name = operator.inputs[2].full_name
+        updates = _cal_tensor_value(node.inputs[2])
+        if (updates != None):
```
Comment (wenbingl) [inline keras2onnx/_builtin.py:2205]: "!= None" ==> "is not None"

File: keras2onnx/_builtin.py (hunk 1 above)
Comment (affranchi) [reply keras2onnx/_builtin.py:2211]: Thank you for the info. Added.

File: keras2onnx/_builtin.py (hunk 2 above)
Comment (affranchi) [reply keras2onnx/_builtin.py]: Done. Thank you for checking.

File: keras2onnx/_builtin.py (hunk 1 above)
Comment (affranchi) [reply keras2onnx/_builtin.py:2211]: Thank you for the info. Added.

**Comment by jiafatom** [comment]
The build error shows (1) flake8 error in linux build.  (2) When you convert, it seems that you need to add Cast node before ScatterND operators because ScatterND need certain type in onnx. Please check onnx specs.

**Comment by affranchi** [comment]
> The build error shows (1) flake8 error in linux build. (2) When you convert, it seems that you need to add Cast node before ScatterND operators because ScatterND need certain type in onnx. Please check onnx specs.

I fixed the two issues but tests are still failing with some weird reasons. One test failed because it tried to run over opset 10 and the other one test failed because it tries with empty input. Could you let me know how to handle the cases?

File: tests/test_layers.py [hunk 3]
```diff
@@ -2580,3 +2580,18 @@ def test_reverseV2(runner):
...
+    updates = np.array([[[1, 1], [1, 1]], [[1, 1], [1, 1]]]).astype(np.float32)
+    tensor = Input(shape=(2,2), name='tensor')
+
+    out = tf.tensor_scatter_nd_update(tensor, indices, updates)
+    model = tf.keras.models.Model(inputs=tensor, outputs=out)
+    onnx_model = keras2onnx.convert_keras(model, model.name)
+
+    tensor_data = np.array([[[6, 5], [6, 6]], [[5, 5], [6, 6]]]).astype(np.float32)
+    expected = model.predict(tensor_data)
+    assert runner(onnx_model.graph.name, onnx_model, tensor, expected)
```
Comment (jiafatom) [inline tests/test_layers.py:2597]: tensor -> tensor_data

File: keras2onnx/_builtin.py [hunk 4]
```diff
@@ -2188,6 +2188,36 @@ def convert_tf_strided_slice(scope, operator, container):
...
+        updates_name = operator.inputs[2].full_name
+        updates = _cal_tensor_value(node.inputs[2])
+        if (updates is not None):
+            updates_arr = np.array(updates)
+            updates_tensor = numpy_helper.from_array(updates_arr, operator.inputs[2].full_name + '_value')
+            container.add_initializer_from_tensor(updates_tensor)
+            updates_name = operator.inputs[2].full_name + '_value'
+
+        oopb.add_node_with_output('ScatterND',
+                                  [operator.inputs[0].full_name] +
```
Comment (jiafatom) [inline keras2onnx/_builtin.py:2213]: Feel that you need add Cast op before ScatterND? check `apply_cast` in our code base.

File: tests/test_layers.py (hunk 3 above)
Comment (affranchi) [reply tests/test_layers.py:2597]: Oops. Revised.

File: keras2onnx/_builtin.py (hunk 4 above)
Comment (affranchi) [reply keras2onnx/_builtin.py:2216]: Added. Thank you for suggestion.

File: tests/test_layers.py [hunk 5]
```diff
@@ -2580,3 +2580,18 @@ def test_reverseV2(runner):
     data = np.random.rand(1, 2, 4).astype(np.float32)
     expected = model.predict(data)
     assert runner('tf_rev_v2', onnx_model, data, expected)
+
+
+@pytest.mark.skipif((is_tensorflow_older_than('1.14.0') or (not is_tf_keras)), reason='old tf version')
+def test_tensor_scatter_update(runner):
```
Comment (jiafatom) [inline tests/test_layers.py:2586]: Add
`@pytest.mark.skipif(get_maximum_opset_supported() < 11,
                    reason="TensorScatterUpdate is not supported before opset 11.")`

Comment (affranchi) [reply tests/test_layers.py:2587]: Added. Thank you for the comment.

**Comment by jiafatom** [comment]
@affranchi Thanks for your contribution!

**Pull request merged**